include-build/roborazzi-gradle-plugin/src/main/java/io/github/takahirom/roborazzi/RoborazziPlugin.kt

```diff
@@ -24,92 +25,98 @@ class RoborazziPlugin : Plugin<Project> {
-      val isVerifyRun = project.objects.property(Boolean::class.java)
-      val isCompareRun = project.objects.property(Boolean::class.java)
-      val isVerifyAndRecordRun = project.objects.property(Boolean::class.java)
-
-      project.gradle.taskGraph.whenReady { graph ->
-        isRecordRun.set(recordTaskProvider.map { graph.hasTask(it) })
-        isVerifyRun.set(verifyTaskProvider.map { graph.hasTask(it) })
-        isVerifyAndRecordRun.set(verifyAndRecordTaskProvider.map { graph.hasTask(it) })
-        isCompareRun.set(compareReportGenerateTaskProvider.map { graph.hasTask(it) })
-      }
+        val verifyTaskProvider =
+          project.tasks.register("verifyRoborazzi$variantSlug", RoborazziTask::class.java) {
+            it.group = VERIFICATION_GROUP
+          }
+        verifyVariants.configure { it.dependsOn(verifyTaskProvider) }
 
-      val testTaskProvider = project.tasks.named("test$testVariantSlug", Test::class.java) { test ->
-//        test.outputs.dir(reportOutputDir)
-//        test.outputs.dir(snapshotOutputDir)
-
-        val roborazziProperties =
-          project.properties.filterKeys { it.startsWith("roborazzi") }
-        val compareReportDir = project.file(RoborazziReportConst.compareReportDirPath)
-
-        test.doFirst {
-          test.systemProperties["roborazzi.test.record"] =
-            isRecordRun.get() || isVerifyAndRecordRun.get()
-          test.systemProperties["roborazzi.test.compare"] = isCompareRun.get()
-          test.systemProperties["roborazzi.test.verify"] =
-            isVerifyRun.get() || isVerifyAndRecordRun.get()
-          test.systemProperties.putAll(roborazziProperties)
-          if (isCompareRun.get()) {
-            compareReportDir.deleteRecursively()
-            compareReportDir.mkdirs()
+        val verifyAndRecordTaskProvider =
+          project.tasks.register("verifyAndRecordRoborazzi$variantSlug", RoborazziTask::class.java) {
+            it.group = VERIFICATION_GROUP
           }
+        verifyAndRecordVariants.configure { it.dependsOn(verifyAndRecordTaskProvider) }
+
+        val isRecordRun = project.objects.property(Boolean::class.java)
+        val isVerifyRun = project.objects.property(Boolean::class.java)
+        val isCompareRun = project.objects.property(Boolean::class.java)
+        val isVerifyAndRecordRun = project.objects.property(Boolean::class.java)
+
+        project.gradle.taskGraph.whenReady { graph ->
+          isRecordRun.set(recordTaskProvider.map { graph.hasTask(it) })
+          isVerifyRun.set(verifyTaskProvider.map { graph.hasTask(it) })
+          isVerifyAndRecordRun.set(verifyAndRecordTaskProvider.map { graph.hasTask(it) })
+          isCompareRun.set(compareReportGenerateTaskProvider.map { graph.hasTask(it) })
         }
+
+        val testTaskProvider = project.tasks.withType(Test::class.java)
+          .matching { it.name == "test$testVariantSlug" }
+          .configureEach { test ->
```

Comment:
Perhaps this function returns Unit, and we are using the testTaskProvider variable as a parameter for dependsOn(). This appears to be the cause of the error we are seeing. 👀